### PR TITLE
Migrate test_get_state_on_brick_unmount

### PR DESCRIPTION
**Pull request by aujjwal-redhat**
Migrate the [test
case](https://github.com/gluster/glusto-tests/blob/HEAD/tests/functional/glusterd/test_get_state_on_brick_unmount.py)

Added get_gluster_state
Added check_if_pattern_in_file

Updates: #292

Fixes: #612

Signed-off-by: Ayush Ujjwal <aujjwal@redhat.com>

### CODE REVIEW

**Comment by aujjwal-redhat** [comment]
I ran the test after commenting the unmount part and check_if_pattern in file's condition i.e. from line 64 to 69:

```js
Table:
 test_get_state_on_brick_unmount
 
+-------------+-------------+--------------------+
| Volume Type | Test Result |  Time taken (sec)  |
+-------------+-------------+--------------------+
|     rep     |     PASS    | 49.922502279281616 |
|     dist    |     PASS    | 49.687578439712524 |
|   dist-rep  |     PASS    |  56.8845272064209  |
|     disp    |     PASS    | 49.80483675003052  |
|  dist-disp  |     PASS    | 54.36140537261963  |
|     arb     |     PASS    | 49.00736880302429  |
|   dist-arb  |     PASS    | 54.31419801712036  |
+-------------+-------------+--------------------+



Framework runtime : 373.44455647468567
```

File: .pylintrc [hunk 1]
```diff
@@ -164,7 +164,8 @@ disable=print-statement,
         F841,
         C901,
         R0915,
-        W0102
+        W0102,
+        C0302
```
Comment (nik-redhat) [inline .pylintrc:168]: Now, what is this for? :sweat_smile:

File: common/ops/gluster_ops/volume_ops.py [hunk 2]
```diff
@@ -863,6 +868,76 @@ def volume_sync(self, hostname: str, node: str, volname: str = 'all'):
 
         return ret
 
+    def get_gluster_state(self, node: str):
```
Comment (nik-redhat) [inline common/ops/gluster_ops/volume_ops.py:871]: We already have a get_state function in gluster_ops.

**Review by nik-redhat — CHANGES_REQUESTED** [review]

**Comment by nik-redhat** [comment]
> I ran the test after commenting the unmount part and check_if_pattern in file's condition i.e. from line 64 to 69:

Why?

File: .pylintrc (hunk 1 above)
Comment (aujjwal-redhat) [reply .pylintrc:168]: It was saying the file has more than 1000 lines. After adding the function the line number went to 1049 so XD

File: common/ops/gluster_ops/volume_ops.py (hunk 2 above)
Comment (aujjwal-redhat) [reply common/ops/gluster_ops/volume_ops.py:871]: Oh ok

**Comment by aujjwal-redhat** [comment]
> > I ran the test after commenting the unmount part and check_if_pattern in file's condition i.e. from line 64 to 69:
> 
> Why?

Discussed in gchat already : |
Basically the brick was coming out to be busy since we don't have multiple disks for bricks.

File: .pylintrc (hunk 1 above)
Comment (nik-redhat) [reply .pylintrc:168]: Okay, see this is a valid point and let the person writing the op add this `pylint: disable` statement individually on top of the ops file. Don't add under exceptions

Comment (aujjwal-redhat) [reply .pylintrc:168]: Ack

**Comment by aujjwal-redhat** [comment]
```js
Table:
 test_get_state_on_brick_unmount
 
+-------------+-------------+--------------------+
| Volume Type | Test Result |  Time taken (sec)  |
+-------------+-------------+--------------------+
|     rep     |     PASS    | 50.80234980583191  |
|     dist    |     PASS    | 51.45233702659607  |
|   dist-rep  |     PASS    | 56.925206422805786 |
|     disp    |     PASS    | 47.06100106239319  |
|  dist-disp  |     PASS    | 54.35770344734192  |
|     arb     |     PASS    | 48.69216442108154  |
|   dist-arb  |     PASS    | 55.98961281776428  |
+-------------+-------------+--------------------+



Framework runtime : 374.1703255176544
```

**Review by srijan-sivakumar — APPROVED** [review]

**Pull request merged**